bot_modules/error_analyse/index.js

```diff
@@ -0,0 +1,162 @@
+  // incase there are multiple errors
+  const exceptions = []
+  stack.groups.forEach((group) => {
+    if (group.exception) {
+      exceptions.push({
+        exception: group.exception.exception,
+        message: group.exception.message,
+        lines: group.lines,
+        packages: [group.stackPackage]
+      })
+    } else {
+      exceptions[exceptions.length - 1].lines.concat(group.lines)
+      exceptions[exceptions.length - 1].packages.push(group.stackPackage)
+    }
+  })
+
+  if (exceptions.length <= 0) {
+    return false
+  }
+
+  // Loop each stacktrace and find any exceptions
+  exceptions.forEach((exception) => {
+    let currentGeyserException = false
+
+    // Check if we found a GeyserMC package in the stack trace
+    for (const stackPackage of exception.packages) {
+      if (stackPackage.name.startsWith('org.geysermc')) {
+        geyserException = true
+        currentGeyserException = true
+        break
+      }
+    }
+
+    // Build the exception title and see if we have a fix
+    const exceptionTitle = exception.exception + ': ' + exception.message
+    const exceptionFix = configExceptionFixes.find(x => exceptionTitle.startsWith(x.error))
+
+    // Check we dont already know about the exception in this log and add it
+    if (exceptionFix && !embed.fields.find(x => x.name === exceptionTitle)) {
+      embed.addField(exceptionTitle, exceptionFix.message)
+    } else if (currentGeyserException) {
+      // Find the first line causing the error in the org.geyser package
+      for (const line of exception.lines) {
+        if (line.stackPackage.name.startsWith('org.geysermc')) {
+          // Get the package url
+          const packageBreakdown = line.stackPackage.name.split('.')
+          let submodule = packageBreakdown[2]
+          if (submodule === 'platform') {
+            submodule = 'bootstrap/' + packageBreakdown[3]
+          }
+
+          // Work out the url for the error
+          const url = `https://github.com/GeyserMC/Geyser/blob/${branch}/${submodule}/src/main/java/${line.stackPackage.name.replace(/\./g, '/')}/${line.source}#L${line.line}`
+
+          // Add a field with the exception details for debugging
+          embed.addField(exceptionTitle, `Unknown fix!\nClass: \`${line.javaClass}\`\nMethod: \`${line.method}\`\nLine: \`${line.line}\`\nLink: [${line.source}#L${line.line}](${url})`)
+          break
+        }
+      }
+    }
+  })
+
+  // Set the base title and description of the embed
+  embed.setTitle('Found errors in the log!')
+  embed.setDescription('See below for details and possible fixes')
+  embed.setColor(0xff0000)
+
+  // If we have no fields set the description acordingly
+  if (embed.fields.length <= 0) {
+    if (geyserException) {
+      embed.setDescription('We don\'t currently have automated responses for the detected errors!')
+    } else {
+      embed.setDescription('The errors you have are not Geyser related or caused!')
```

Comment:
Geyser could probably cause other errors because of conflicts, so maybe say they _probably_ aren't related but could be?